src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,50 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[MAX_SIZE];
+        this.values = (V[]) new Object[MAX_SIZE];
+        this.size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findIndex(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            if (size < MAX_SIZE) {
+                keys[size] = key;
+                values[size] = value;
+                size++;
+            } else {
+                throw new IllegalStateException("Storage is full");
+            }
+        }
```

Comment:
Generally it's considered that less tabulation levels we have the more readable becomes code.
Below you may see same logic but with slightly redesigned `if` blocks
```suggestion
        int index = findIndex(key);
        if (index != -1) {
            values[index] = value;
            return;
        }
        if (size > MAX_SIZE) {
            throw new IllegalStateException("Storage is full");
        } 
        keys[size] = key;
        values[size] = value;
        size++;            
```

Reply:
Your logic is clearer than mine. I completely agree with this. I will take your comments into account in the future. Thank you, your comment is invaluable to me.